src/field_info.rs

```diff
@@ -297,3 +303,41 @@ impl SetterSettings {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct Transform {
+    pub params: Vec<(syn::Pat, syn::Type)>,
+    pub body: syn::Expr,
+}
+
+fn parse_transform_closure(expr: &syn::Expr) -> Result<Transform, Error> {
+    let closure = match expr {
+        syn::Expr::Closure(closure) => closure,
+        _ => return Err(Error::new_spanned(expr, "Expected closure")),
+    };
+    if let Some(kw) = &closure.asyncness {
+        return Err(Error::new(kw.span, "Transform closure cannot be async"));
+    }
+    if let Some(kw) = &closure.capture {
+        return Err(Error::new(kw.span, "Transform closure cannot capture variables"));
```

Comment:
Nit: This error message is a bit confusing at first, because syn calls it `capture` but from the user's view the more natural term is `move`. It's not that bad, because after thinking about it for 10 seconds you can understand what it means, because `rustc` will mark the `move`, and because there is no reason to put `move` on that closure in the first place. Still - I will change that error message if this PR gets merged before you do it.

Reply:
Since the user cannot capture any variables anyway, my motivation is that any such keywords would not make sense either way.